src/transformers/pipelines/base.py

```diff
@@ -866,7 +866,7 @@ def forward(self, model_inputs, **forward_params):
                 model_inputs["training"] = False
                 model_outputs = self._forward(model_inputs, **forward_params)
             elif self.framework == "pt":
-                with torch.no_grad():
+                with torch.inference_mode():
                     model_inputs = self._ensure_tensor_on_device(model_inputs, device=self.device)
                     model_outputs = self._forward(model_inputs, **forward_params)
                     model_outputs = self._ensure_tensor_on_device(model_outputs, device=torch.device("cpu"))
```

Comment:
```suggestion
                if version.parse(torch.__version__) >= version.parse("1.9.0"):     
                    with torch.inference_mode():                                 
                        model_inputs = self._ensure_tensor_on_device(model_inputs, device=self.device)  
                        model_outputs = self._forward(model_inputs, **forward_params)                     
                        model_outputs = self._ensure_tensor_on_device(model_outputs, device=torch.device("cpu"))  
                else:                                                                                               
                    with torch.no_grad():                                          
                        model_inputs = self._ensure_tensor_on_device(model_inputs, device=self.device)  
                        model_outputs = self._forward(model_inputs, **forward_params)                     
                        model_outputs = self._ensure_tensor_on_device(model_outputs, device=torch.device("cpu")) 
```

Reply:
Don't forget to add `from packaging import version` at the top of the file.

Reply:
Or to avoid code duplication:
```
inference_context = torch.inference_mode if version.parse(torch.__version__) >= version.parse("1.9.0") else torch.no_grad

with inference_context():
   xxx
```

Reply:
Done changes @Narsil